lib/collection.js

```diff
@@ -1369,9 +1368,16 @@ Collection.prototype.count = deprecate(function(query, options, callback) {
   query = args.length ? args.shift() || {} : {};
   options = args.length ? args.shift() || {} : {};
 
-  return executeLegacyOperation(this.s.topology, count, [this, query, options, callback]);
+  if (typeof options === 'function') (callback = options), (options = {});
+  options = options || {};
+
+  return executeOperation(
```

Comment:
Is this a change in behavior?

Reply:
Hmm, I'm not sure how it would be? Do you have an idea? The `EstimatedDocumentCountOperation` just temporarily supports a `query` to provide support for the deprecated `count` command here

Reply:
Ah, I didn't catch the `query`, but that makes sense.